### PR TITLE
Add Happiness Table

### PR DESCRIPTION
**Pull request by zachseidner1**
- Add Happiness Table
- Currently is in colored variation, use commit acd1a2b0f4c709b3ed934fcf175d5e736966049e for the original version
- Has its own networking
- Doesn't support going across months right now, not sure if this is something we want to add.

### CODE REVIEW

File: src/components/HappinessTable.tsx [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+  endDate?: string;
+}) {
+  const { api } = useApi();
+  const [selectedHappiness, setSelectedHappiness] = useState<
+    Happiness | undefined
+  >(undefined);
+  // open modal when user selects new happiness
+  useEffect(() => {
+    if (selectedHappiness) {
+      window.HSOverlay.open(document.querySelector("#view-happiness"));
```
Comment (jonathanjma) [inline src/components/HappinessTable.tsx:34]: should we also mark the post as read when opening the modal or should that only be for the feed

File: src/components/HappinessTable.tsx [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+/**
+ *
+ * @param startDate a string in the YYYY-MM-DD format representing the start of
+ * the happiness date range
+ * @param endDate a string in the YYYY-MM-DD format representing the end of
+ * the happiness date range (defaults to today if unspecified)
+ * @param groupId the associated group ID for the happiness table
+ * @returns
+ */
+export default function HappinessTable({
```
Comment (jonathanjma) [inline src/components/HappinessTable.tsx:18]: is the table responsive in the horizontal direction? to view the entire table requires me to scroll left, which puts the sidebar out of view

File: src/components/HappinessTable.tsx [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+                {date.toLocaleDateString("en-us", { day: "numeric" })}
+              </th>
+            ))}
+          </tr>
+        </thead>
+        <tbody className="divide-x-1 divide-y-1 divide-gray-200 border-b bg-white">
+          {group.users.map((user) => (
+            <tr>
+              <td className="text-gray-900 whitespace-nowrap border-l border-r border-gray-200 px-6 py-4 text-sm font-medium">
+                {user.username}
```
Comment (jonathanjma) [inline src/components/HappinessTable.tsx:108]: just curious were we still thinking about adding pfps next to the usernames?

File: src/components/HappinessTable.tsx (hunk 2 above)
Comment (zachseidner1) [reply src/components/HappinessTable.tsx:18]: If you wrap the table in a parent of fixed width it scrolls properly. I also made the first columns sticky so as you scroll you can see whose happiness is whose.

File: src/components/HappinessTable.tsx (hunk 1 above)
Comment (zachseidner1) [reply src/components/HappinessTable.tsx:34]: yes, just did that

**Comment by jonathanjma** [comment]
i think you forgot to push your updated code?

**Review by jonathanjma — APPROVED** [review]
nice work :-)

**Pull request merged**